File: engine/transpile/src/test/resources/golden/SYK006/HOST_倉庫コード.py
"""SYK006 のデータ部レコード HOST-倉庫コード を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class HOST_倉庫コード:
    """01 HOST-倉庫コード 総バイト長 4。"""

    LENGTH = 4

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(4)
        else:
            self._data = bytearray(data)
            if len(self._data) < 4:
                self._data.extend(b"\x00" * (4 - len(self._data)))

    @property
    def data(self):
        return self._data

    # HOST-倉庫コード
    def get_HOST_倉庫コード(self):
        return decode_alphanumeric(self._data, 0, 4)
    def set_HOST_倉庫コード(self, value):
        encode_alphanumeric(self._data, 0, 4, value)

File: src/engine/transpile/src/test/resources/golden/SYK007/HOST_倉庫コード.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK007 のデータ部レコード HOST-倉庫コード を逐語対訳した自動生成コード。 */
public final class HOST_倉庫コード {

    /** 01 HOST-倉庫コード 総バイト長 4。 */
    public static final int LENGTH = 4;

    private final byte[] data;

    public HOST_倉庫コード() {
        this.data = new byte[4];
    }

    public HOST_倉庫コード(byte[] source) {
        this.data = Arrays.copyOf(source, 4);
    }

    public byte[] data() {
        return data;
    }

    // HOST-倉庫コード
    public String get_HOST_倉庫コード() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 4);
    }
    public void set_HOST_倉庫コード(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 4, value);
    }
}
